internalsdk/pro/proxy.go

```diff
@@ -1,115 +0,0 @@
-package pro
-
-import (
-	"bytes"
-	"compress/gzip"
-	"encoding/json"
-	"io"
-	"io/ioutil"
-	"net/http"
-	"net/http/httputil"
-	"strconv"
-	"strings"
-	"time"
-
-	"github.com/getlantern/lantern-client/internalsdk/common"
-	"github.com/getlantern/lantern-client/internalsdk/protos"
-)
-
-type proxyTransport struct {
-	// Satisfies http.RoundTripper
-}
-
-func (pt *proxyTransport) processOptions(req *http.Request) *http.Response {
-	resp := &http.Response{
-		StatusCode: http.StatusOK,
-		Header: http.Header{
-			"Connection": {"keep-alive"},
-			"Via":        {"Lantern Client"},
-		},
-		Body: ioutil.NopCloser(strings.NewReader("preflight complete")),
-	}
-	if !common.ProcessCORS(resp.Header, req) {
-		return &http.Response{
-			StatusCode: http.StatusForbidden,
-		}
-	}
-	return resp
-}
-
-func (pt *proxyTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
-	if req.Method == "OPTIONS" {
-		// No need to proxy the OPTIONS request.
-		return pt.processOptions(req), nil
-	}
-	origin := req.Header.Get("Origin")
-	// Workaround for https://github.com/getlantern/pro-server/issues/192
-	req.Header.Del("Origin")
-	resp, err = NewHTTPClient(http.DefaultTransport, time.Duration(time.Second*30)).Do(req)
```

Comment:
This is where `proxyTransport` may have _thought_ it was proxying requests.  In reality, the client created by `NewHTTPClient` would send the requests directly.

Reply:
Yeah, that looks wrong. We fortunately weren't using this code though (thanks for removing it!)